objects/src/notes/note_tag.rs

```diff
@@ -19,72 +24,17 @@ impl NoteTag {
     // --------------------------------------------------------------------------------------------
 
     /// Returns a new [NoteTag] instantiated from the specified account ID.
-    ///
-    /// The tag is constructed as follows:
-    /// - For local execution, the two most significant bits are set to 0b00, the following 16 bits
-    ///   are set to the 16 most significant bits of the account ID, and the remaining 14 bits are
-    ///   set to 0.
-    /// - For network execution, the two most significant bits are set to 0b10 and the remaining
-    ///   bits are set to the 30 most significant bits of the account ID.
-    pub fn from_account_id(
-        account_id: AccountId,
-        execution: NoteExecutionMode,
-    ) -> Result<Self, NoteError> {
```

Comment:
I think this method worked as intended - i.e., it constructed tags for notes indented to be executed by specific accounts (e.g., tags with prefixes `00` and `101`).

What we probably should do, is add another method - e.g., something like `from_use_case()`. This method could look like this:
```Rust
pub fn for_use_case(
    use_case_type: u16,
    payload: u16,
    execution:NoteExecutionMode
) -> Result<Self, NoteError>
```
This constructor would create notes intended for consumption by any account (i.e., tag prefixes `01`, `101`, and `11`).

The validation that we'd need to do is that `use_case_type` is actually just 14 bits (we could also create a new type for this). Alternatively, we could try to create a use case enum:
```Rust
pub enum NoteUseCase {
    Swap(u8, u8), // internal values are the first 8 bits of the from/to faucet IDs
}
```
But the question then is how to make `NoteUseCase` enum extensible.